gtk/src/toga_gtk/widgets/base.py

```diff
@@ -68,9 +72,9 @@ def set_tab_index(self, tab_index):
     ######################################################################
 
     def set_bounds(self, x, y, width, height):
-        # No implementation required here; the new sizing will be picked up
-        # by the box's allocation handler.
-        pass
+        # If the bounds have changed, we need to queue a resize on the container
+        if self.container:
+            self.container.make_dirty()
```

Comment:
In gtk `Widget.set_bounds`.

Reply:
This one definitely seems like a hangover that didn't hurt, but doesn't help either. When `set_bounds()` is invoked, it's positioning a specific widget in response to a layout; in the GTK context, that *must* be in response to a layout. I'll push an update to drop this.